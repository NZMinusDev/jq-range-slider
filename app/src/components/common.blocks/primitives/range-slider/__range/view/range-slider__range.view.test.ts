import RangeSliderRangeView, { DEFAULT_OPTIONS } from "./range-slider__range.view";

import {
  InstancePropsExpecter,
  testInit,
  testInitDEFAULT_OPTIONS,
  testGetter,
  testSetter,
  DifferentArguments,
} from "@utils/devTools/tools/UnitTestingHelper";

const viewPropertiesExpecter: InstancePropsExpecter<
  ConstructorParameters<typeof RangeSliderRangeView>,
  RangeSliderRangeView
> = function ({ instance }) {};

const differentOptionsArg: DifferentArguments<Parameters<
  typeof RangeSliderRangeView.prototype.setOptions
>> = {
  fullOptionalArguments: [[{ isConnected: true }]],
};

testInitDEFAULT_OPTIONS(RangeSliderRangeView, [DEFAULT_OPTIONS], viewPropertiesExpecter);

testInit({
  Creator: RangeSliderRangeView,
  differentConstructorArgs: {
    validRequiredArguments: [[]],
    ...(differentOptionsArg as DifferentArguments<
      ConstructorParameters<typeof RangeSliderRangeView>
    >),
  },
  instancePropsExpecter: viewPropertiesExpecter,
  propsToSet: new Map().set("_options", 1),
});

testGetter({
  Creator: RangeSliderRangeView,
  constructorArgs: [],
  instancePropsExpecter: viewPropertiesExpecter,
  methodOfInstanceToTest: {
    methodReference: RangeSliderRangeView.prototype.getOptions,
    expecter: ({ mock, passedArgs, instance }) => {},
    returns: "_options",
  },
});
testSetter({
  Creator: RangeSliderRangeView,
  constructorArgs: [],
  instancePropsExpecter: viewPropertiesExpecter,
  methodOfInstanceToTest: {
    methodReference: RangeSliderRangeView.prototype.setOptions,
    expecter: ({ mock, passedArgs, instance }) => {},
    differentArguments: differentOptionsArg,
  },
  propsToSet: new Map().set("_options", 0),
  resetPropsTo: new Map().set("_options", DEFAULT_OPTIONS),
});
