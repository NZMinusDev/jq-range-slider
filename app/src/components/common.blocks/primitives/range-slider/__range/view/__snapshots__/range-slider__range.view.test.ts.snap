// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM manipulation renders correctly 1`] = `
<DocumentFragment>
  <!---->
  <div
    class="range-slider__range"
    style=""
  />
  <!---->
</DocumentFragment>
`;
